### DISCUSSION TITLE
Adding Additional Data Points for CLV

### DISCUSSION THREAD

**Discussion by acuson** [pymc-labs/pymc-marketing#638] [Q&A] [answered]
Hi - for the CLV element of this model, I reviewed the section of the data requirements, and it looks like it's only concrete numbers related to purchases, customer age, etc. Is is possible to include other customer attributes in this model to help predict CLV (such as age group, geographic location, income range, etc.)? Are there any concerns adding these additional data points within the model?

**Comment by ColtAllen** [comment, marked as answer]
Hey @acuson,

[`ParetoNBDModel`](https://www.pymc-marketing.io/en/stable/api/generated/pymc_marketing.clv.models.pareto_nbd.ParetoNBDModel.html#pymc_marketing.clv.models.pareto_nbd.ParetoNBDModel) currently supports time invariant covariates, and other models will support this in the future.

You would parametrize the model as shown:

```python
ParetoNBDModel(data, model_config = {
    "purchase_covariate_cols": ['covariate_col_name1', 'covariate_col_name2', ...],
    "dropout_covariate_cols": ['covariate_col_name1', 'covariate_col_name2', ...]
    }
)
```

I'm working on a notebook with a covariate example in fact, and will be posting opening that PR within the next few days.

Many of the covariate examples you mentioned are categorical. One-hot encoding would make for more interpretable results, but watch out for the dummy variable trap:

[Encoding Categorical Variables: One-hot vs Dummy Encoding](https://towardsdatascience.com/encoding-categorical-variables-one-hot-vs-dummy-encoding-6d5b9c46e2db)

**Reply by ColtAllen** [reply]
The new [Pareto/NBD Notebook](https://www.pymc-marketing.io/en/stable/notebooks/clv/pareto_nbd.html) contains a simple covariate example in the very last section.
